core/server/master/src/main/java/alluxio/master/metastore/caching/CachingInodeStore.java

```diff
@@ -690,9 +688,6 @@ public void removeEdge(Edge edge) {
         // Skip caching if the cache is full or someone else is already caching.
         return mEdgeCache.getChildIds(inodeId).values();
       }
-      if (entry.mChildren != null) {
-        return entry.mChildren.values();
-      }
```

Comment:
We can't trust the cached value here?

Reply:
we can, but we already checked it 7 lines earlier so this line does nothing